grafana/writers-toolkit
# add deploy preview workflow

for https://github.com/grafana/website/issues/10101

This PR deploys a publicly available preview of the output of the `make docs` command for pull requests. The preview is then deleted when the PR is merged or closed.

### To test
1. Create a branch from this PR
1. Open a PR
1. Test the GitHub comment link
1. Test the link in the GitHub commit checks area
1. Push a new test commit
1. Once the deploy preview is updated, make sure the commit is reflected: 
```
curl -Is https://deploy-preview-writers-toolkit-{replace-with-pr}-zb444pucvq-vp.a.run.app | grep build
```
The output should be the same commit sha as the commit you pushed.
1. Again test the GitHub link (you should see the comment has been edited)
1. Close the PR
1. Confirm the comment is updated with "Deploy preview deleted"
2. Confirm the preview is actually deleted - it should 404.

Before merge todo:
- [x] change clone ref branch
- [x] change workflow ref branch
